### ISSUE TITLE
How to get Owner.XXX fields for custom objects? fflib_QueryFactory

### ISSUE THREAD

**Issue by cropredy** [apex-enterprise-patterns/fflib-apex-common#131]
Assume we have a selector on a custom object `Foo__c`

The following does not work:

```
public List<Foo__c> selectByIdWithOwner(set<ID> idSet>) {
   fflib_QueryFactory fooQF = newQueryFactory()
                                     .selectFields(new List<String>{'Owner.Alias'});
}
```

Result is error `EXCEPTION_THROWN [112]|fflib_QueryFactory.InvalidFieldException: Invalid field 'Alias' for object 'Group'`

What happens is the method `selectFields` in `fflib_QueryFactory` calls `getFieldToken('Owner.Alias')` and that method does a describe on `OwnerId` and comes back with `lastSObjectType` = `Group.SObjectType` (!?!) and a comment `//if it's polymorphic doesn't matter which one we get`

But it does matter as there is no field `Group.Alias` - only `User.Alias`.  Since custom objects can be members of queues, the describe is returning, by chance, via `getReferenceTo()[0]` the Group (i.e. Queue) and not the intended `User`.

AFAIK, there is no way around this when using the `fflib_QueryFactory` and one has to resort to the legacy  `String.format(...)` query construction (and thus you end up with a mix of styles for query construction in your code base)

One possible suggestion would be to allow for the same syntax as in [polymorphic Owner formula fields](https://developer.salesforce.com/blogs/developer-relations/2013/06/cross-object-owner-fields-a-powerful-new-formula-option.html) 

```
public List<Foo__c> selectByIdWithOwner(set<ID> idSet>) {
   fflib_QueryFactory fooQF = newQueryFactory()
                                  .selectFields(new List<String>{'Owner:User.Alias'});
}
```

 and

```
public List<Foo__c> selectByIdWithOwner(set<ID> idSet>) {
   fflib_QueryFactory fooQF = newQueryFactory();

   new UsersSelector().configureQueryFactoryFields(fooQF,'Owner:User');
}
```

**Comment by afawcett** [comment]
Hmmm lets see if Chris Peterson has any thoughts on this, i'll tweet him this issue.

**Comment by capeterson** [comment]
This is a very valid complaint. I'll see if I can find time to get a PR together to improve it to allow use of any relationship.

**Comment by cropredy** [comment]
Thanks Chris, I appreciate whatever you can do.  

--------------
Eric Kintzer

**Comment by cropredy** [comment]
Thanks Andy

--------------
Eric Kintzer

**Comment by briancodey** [comment]
Is there any available enhancement on this?

**Comment by daveespo** [comment]
This was likely fixed on #137 (actually merged in PR #402 due to the need to rebase)
